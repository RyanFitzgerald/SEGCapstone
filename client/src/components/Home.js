import React from 'react';

class Home extends React.Component {
  componentDidMount() {
    // Set title
    document.title = 'Home | Renovaction';

    // Update active tab
    this.props.setActiveTab(1);
  }

  render() {
    return (
      <div className="content">
        <div className="row">
          <div className="column">
            <h2 className="card-title">Quick Actions</h2>
            <div className="card">
              Content to be determined
            </div>
          </div>
        </div>
        <div className="row">
          <div className="md-6 column">
            <h2 className="card-title">Projects Summary</h2>
            <div className="card">
              Content to be determined
            </div>
          </div>
          <div className="md-6 column">
            <h2 className="card-title">Clients Summary</h2>
            <div className="card">
              Content to be determined
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Home;