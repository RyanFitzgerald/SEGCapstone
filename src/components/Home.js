import React, { Component, PropTypes } from 'react';
import {Link} from 'react-router-dom';

class Home extends Component {

    constructor(props) {
        super(props);
    }

    componentDidMount() {
        // Update page title
        document.title = 'Dashboard';

        // Update Header Title
        this.props.setHeaderTitle('Dashboard');

        // Update Active Tab
        this.props.setActiveTab(1);
    }


    render() {
        return (
            <div className="home">
                <div id="dashboard-actions">
                    <div className="row">
                        <div className="small-12 columns">
                            <h2>Quick Actions</h2>
                            <div className="row small-up-1 medium-up-2 large-up-4">
                                <div className="column">
                                    <Link className="quick-action rounded" to="/projects">
                                        <div className="qa-icon">
                                            <i className="fa fa-search" aria-hidden="true"></i>
                                        </div>
                                        {/* <!-- End qa-icon --> */}
                                        <div className="qa-text">
                                            <span>Projects Overview</span>
                                        </div>
                                        {/* <!-- End qa-text --> */}
                                    </Link>
                                </div>
                                <div className="column">
                                    <Link className="quick-action rounded" to="/clients">
                                        <div className="qa-icon">
                                            <i className="fa fa-search" aria-hidden="true"></i>
                                        </div>
                                        {/* <!-- End qa-icon --> */}
                                        <div className="qa-text">
                                            <span>Clients Overview</span>
                                        </div>
                                        {/* <!-- End qa-text --> */}
                                    </Link>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                {/* <!-- End dashboard-actions --> */}

                <div id="dashboard-other">
                    <div className="row">
                        <div className="small-12 large-6 columns">
                            <h2>Project Overview</h2>
                            <div id="dashboard-overview" className="rounded">
                                Content to be determined
                            </div>
                            {/* <!-- End dashboard-overview --> */}
                        </div>
                        <div className="small-12 large-6 columns">
                            <h2>Client Overview</h2>
                            <div id="dashboard-notifications" className="rounded">
                                Content to be determined
                            </div>
                            {/* <!-- End dashboard-notifications --> */}
                        </div>
                    </div>
                </div>
                {/* <!-- End dashboard-other --> */}

            </div>
        );
    }
}

Home.propTypes = {
    setActiveTab: PropTypes.func.isRequired,
    setHeaderTitle: PropTypes.func.isRequired,
    history: PropTypes.any
};

export default Home;
