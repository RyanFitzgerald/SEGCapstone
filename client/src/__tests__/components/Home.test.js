import React from 'react';
import Home from '../../Components/Home';

describe('<Home />', () => {
  it('should render correctly', () => {
    // Create mock functions / spies
    const setActiveTabMock = jest.fn();

    // Shallow render it
    const wrapper = shallow(
      <Home setActiveTab={setActiveTabMock}/>
    );

    // Compare to old snapshot
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    // Check set active tab is called and with correct value
    expect(setActiveTabMock).toHaveBeenCalledTimes(1);
    expect(setActiveTabMock.mock.calls[0][0]).toBe(1);
  });
});
