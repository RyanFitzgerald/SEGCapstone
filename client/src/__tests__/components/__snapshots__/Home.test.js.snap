// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render correctly 1`] = `
<div
  className="content"
>
  <div
    className="row"
  >
    <div
      className="column"
    >
      <h2
        className="card-title"
      >
        Quick Actions
      </h2>
      <div
        className="card"
      >
        <ul
          className="quick-actions"
        >
          <li>
            <Link
              replace={false}
              to="/projects"
            >
              <i
                aria-hidden="true"
                className="fa fa-building"
              />
               View Projects
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/clients"
            >
              <i
                aria-hidden="true"
                className="fa fa-address-card"
              />
               View Clients
            </Link>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
